### PR TITLE
[FLOC-3216] Added scripts to download and analyze failed tests

### PR DESCRIPTION
**Pull request by thepeak99**
A quick hack in Python and Ansible to download buildbot's failed jobs and parse their output

[Review on Reviewable](https://reviewable.io/reviews/clusterhq/build.clusterhq.com/144)

### CODE REVIEW

File: scripts/analyze_failed_jobs/action_plugins/parsejobs.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import glob, pickle, buildbot, re, sys, cStringIO as StringIO
```
Comment (jml) [inline scripts/analyze_failed_jobs/action_plugins/parsejobs.py]: Please add a module docstring describing how this is used and what it's used for.

File: scripts/analyze_failed_jobs/action_plugins/parsestdouts.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import re, glob, bz2
```
Comment (jml) [inline scripts/analyze_failed_jobs/action_plugins/parsestdouts.py]: Likewise here. Please include an example of what's used and an example of what it's used for.

File: scripts/analyze_failed_jobs/playbook.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+- hosts: build.clusterhq.com
+  become: true
+  vars: 
+    buildbot_dir: /var/lib/docker/vfs/dir/91ca9fb14be23372a04b364447224e6d8058dbac4ea51905c7c6e42fbb06f1bd
```
Comment (jml) [inline scripts/analyze_failed_jobs/playbook.yml]: Is this going to need to be changed after we redeploy the buildbot? If so, please add a comment.

File: scripts/analyze_failed_jobs/action_plugins/parsejobs.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+from ansible.utils import template
+from ansible.runner.return_data import ReturnData
+
+class ActionModule:
+    def __init__(self, runner):
+        self.runner = runner
+        
+        
+    def run(self, conn, tmp, module_name, module_args, inject, complex_args=None, **kwargs):
```
Comment (jml) [inline scripts/analyze_failed_jobs/action_plugins/parsejobs.py]: I think it's worth documenting what each of these parameters mean.

File: scripts/analyze_failed_jobs/action_plugins/parsestdouts.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    'flocker-acceptance-vagrant-centos-7-zfs',
+    'flocker-installed-package-vagrant-centos-7'
+]
+
+class ActionModule:
+    def __init__(self, runner):
+        self.runner = runner
+        
+        
+    def run(self, conn, tmp, module_name, module_args, inject, complex_args=None, **kwargs):
```
Comment (jml) [inline scripts/analyze_failed_jobs/action_plugins/parsestdouts.py]: Likewise, please document each of these parameters.

File: scripts/analyze_failed_jobs/action_plugins/parsejobs.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        jobs = glob.glob('work/*/*')
+
+        out = StringIO.StringIO()
+
+        count = 0
+        count_failed = 0
+
+        for job in jobs:
+            foo, builder, job_num = job.split('/')
```
Comment (jml) [inline scripts/analyze_failed_jobs/action_plugins/parsejobs.py]: Just use `_` for an unused variable, rather than `foo`

**Comment by jml** [comment]
After my request for a quick and dirty script I get to indulge in a round of hypocrisy and ask you to clean it up :wink: 

In addition to the comments above, you'll need to fix up some lint before this lands. https://travis-ci.org/ClusterHQ/build.clusterhq.com/jobs/84085059 has the details, or you could run `flake8` on your local machine.

File: scripts/analyze_failed_jobs/action_plugins/parsejobs.py (hunk 1 above)
Comment (thepeak99) [inline scripts/analyze_failed_jobs/action_plugins/parsejobs.py]: Well that's an Ansible module, its not meant to be used directly, its called by Ansible. Maybe I should add a README.md to explain how to run the playbook with Ansible?

File: scripts/analyze_failed_jobs/action_plugins/parsejobs.py (hunk 4 above)
Comment (thepeak99) [inline scripts/analyze_failed_jobs/action_plugins/parsejobs.py]: Same as above, it's the Ansible API... as you see I'm almost not using any of those parameters, this is run by Ansible when the module is called. Maybe I could point people to Ansible docs?

File: scripts/analyze_failed_jobs/action_plugins/parsestdouts.py (hunk 5 above)
Comment (thepeak99) [inline scripts/analyze_failed_jobs/action_plugins/parsestdouts.py]: I could document though the Ignored Builders constant... although its pretty self-explanatory.
To make it "proper" I should have used an Ansible parameter, but you requested a quick and dirty hack

File: scripts/analyze_failed_jobs/action_plugins/parsejobs.py (hunk 1 above)
Comment (tomprince) [inline scripts/analyze_failed_jobs/action_plugins/parsejobs.py]: Even if it an ansible module, it should have a docstring describing what it is, and what it is used for.

File: scripts/analyze_failed_jobs/playbook.yml (hunk 3 above)
Comment (tomprince) [inline scripts/analyze_failed_jobs/playbook.yml]: `docker cp` might be a better solution to hard-coding the path?

File: scripts/analyze_failed_jobs/action_plugins/parsejobs.py (hunk 1 above)
Comment (jml) [inline scripts/analyze_failed_jobs/action_plugins/parsejobs.py]: Yes, please add a `README.md` as you describe.

However, Tom is right, please add a short docstring. 

```
"""
Ansible module for finding failed buildbot jobs.

Stores results in ``out/count.csv``, which has columns for $FOO, $BAR, $BAZ.
"""
```

File: scripts/analyze_failed_jobs/action_plugins/parsejobs.py (hunk 4 above)
Comment (jml) [inline scripts/analyze_failed_jobs/action_plugins/parsejobs.py]: I think that's fine as-is then.

File: scripts/analyze_failed_jobs/action_plugins/parsestdouts.py (hunk 5 above)
Comment (jml) [inline scripts/analyze_failed_jobs/action_plugins/parsestdouts.py]: No, I'm OK with it as-is. Thanks.

**Comment by jml** [comment]
Thanks for the lint fixes. Still to go:

* Module docstrings
* README.md (it can be short, we can make it better later)
* Optional: use docker cp

And we're good to merge. Thanks!

File: scripts/analyze_failed_jobs/action_plugins/parsejobs.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
+"""
+This Ansible module reads Buildbot jobs, identifies failed ones, and outputs
```
Comment (jml) [inline scripts/analyze_failed_jobs/action_plugins/parsejobs.py]: There's trailing whitespace here that's failing the linter.

File: scripts/analyze_failed_jobs/action_plugins/parsestdouts.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
+"""
+Ansible module for finding failed buildbot jobs.
+
+Stores results in ``out/test-count.csv``, which has columns for builder, test and failure count.
```
Comment (jml) [inline scripts/analyze_failed_jobs/action_plugins/parsestdouts.py]: Please wrap this line to <= 79 characters. It's failing the linter.

**Comment by jml** [comment]
Great, thanks.  A couple of minor changes are necessary to prevent Travis from failing.

**Pull request merged**